src/types/metaFactory.ts

```diff
@@ -22,18 +24,30 @@ export type MetaFactory = Contract & {
   gap?: string
   standard: NFTStandard; // which type of factories will this metaFactory create
   autoApprove: boolean;
+  typeMetadata?: TypeMetadata;
 }
 
 export type MetaFactoryType = {
   newContractCreatedEvent: string,
   creationEventToNftFactory?: (event: ethers.Event, autoApprove: boolean, factoryMetadata?: unknown) => NftFactory
-  metadataAPI?: (events: ethers.Event[], clients: Clients) => Promise<any>
+  metadataAPI?: (events: ethers.Event[], clients: Clients) => Promise<any>,
+  metadataAccountToNftFactory?: (metadata: Metadata, metaFactory: MetaFactory) => NftFactory
```

Comment:
let's try merge creationEventToNftFactory and metadataAccountToNftFactory so that they conform to the same interface.

i guess this means that instead of accepting a specific ethers.Event[] for the creationEvent it should be more generic, like `creationEventMetadataToNftFactory`

So it could look like

`creationMetadataToNftFactory?: (creationData: unknown, autoApprove: boolean, factoryMetadata?: unknown) => NftFactory `